java/shared/src/test/java/com/squareup/subzero/shared/SubzeroUtilsTest.java

```diff
@@ -139,6 +189,38 @@ public void setUp() {
     assertArrayEquals(expected, childKey.getPubKey());
   }
 
+  @Test public void testDerivePublicKeyNegativeIndex() {
+    DeterministicKey rootKey = HDKeyDerivation.createMasterPrivateKey(new byte[32]);
+    Path invalidPath = Path.newBuilder().setIsChange(false).setIndex(-1).build();
+
+    Exception exception = assertThrows(IllegalStateException.class, () -> {
+      SubzeroUtils.derivePublicKey(rootKey, invalidPath);
+    });
+
+    assertEquals("index should be between 0 and 2^31-1", exception.getMessage());
+  }
+
+  @Test public void testDerivePublicKeyMaxIndex() {
+    DeterministicKey rootKey = HDKeyDerivation.createMasterPrivateKey(new byte[32]);
+    // Integer.MAX_VALUE is 2^31-1.
+    Path path = Path.newBuilder().setIsChange(false).setIndex(Integer.MAX_VALUE).build();
+
+    DeterministicKey derivedKey = SubzeroUtils.derivePublicKey(rootKey, path);
+    assertEquals(33, derivedKey.getPubKey().length);
```

Comment:
If we have a hard-coded root key (looks like we're using an all-zeroes key here?), can we hard-code the expected derived key here and check for equality?